Makefile

```diff
@@ -32,7 +35,7 @@ all: generate unit
 generate: smithy-generate gen-config-asserts gen-repo-mod-replace tidy-modules-. add-module-license-files
 
 smithy-generate:
-	cd codegen && ./gradlew clean build -Plog-tests && ./gradlew clean
+	cd codegen && ./gradlew clean build -Plog-tests
```

Comment:
I thought this was hear to avoid having build directories being present in the repo and being iterated over?

Reply:
The eachmodule now automatically ignores the `/codegen` since there is not really any case we want any module in there to be evaluated.

Reply:
added new `smithy-build` that don't clean `smith-generate` will now do the clean after the build.